src/setup_node_env/babel_register/register.js

```diff
@@ -46,26 +46,12 @@ var ignore = [
 
   // ignore paths matching `/canvas/canvas_plugin/`
   /[\/\\]canvas[\/\\]canvas_plugin[\/\\]/,
-];
 
-if (global.__BUILT_WITH_BABEL__) {
-  // when building the Kibana source we replace the statement
-  // `global.__BUILT_WITH_BABEL__` with the value `true` so that
-  // when @babel/register is required for the first time by users
-  // it will exclude kibana's `src` directory.
-  //
-  // We still need @babel/register for plugins though, we've been
-  // building their server code at require-time since version 4.2
-  // TODO: the plugin install process could transpile plugin server code...
-  ignore.push(resolve(__dirname, '../../../src'));
-} else {
-  ignore.push(
-    // ignore any path in the packages, unless it is in the package's
-    // root `src` directory, in any test or __tests__ directory, or it
-    // ends with .test.js, .test.ts, or .test.tsx
-    /[\/\\]packages[\/\\](eslint-|kbn-)[^\/\\]+[\/\\](?!src[\/\\].*|(.+[\/\\])?(test|__tests__)[\/\\].+|.+\.test\.(js|ts|tsx)$)(.+$)/
-  );
-}
+  // ignore any path in the packages, unless it is in the package's
+  // root `src` directory, in any test or __tests__ directory, or it
+  // ends with .test.js, .test.ts, or .test.tsx
+  /[\/\\]packages[\/\\](eslint-|kbn-)[^\/\\]+[\/\\](?!src[\/\\].*|(.+[\/\\])?(test|__tests__)[\/\\].+|.+\.test\.(js|ts|tsx)$)(.+$)/,
```

Comment:
@spalger should we add an extra line to always ignore code on `src` and `xpack` in case we require by mistake `setup_node_env/index.js`?

Reply:
No need, this module isn't excluded from the build by [the changes to the copy_source_task](https://github.com/spalger/kibana/blob/remove/babel-register-production/src/dev/build/tasks/copy_source_task.ts#L40-L41) so there's no way to accidentally get it into production.